### PR TITLE
feat: remove restriction of only checking default value

### PR DESCRIPTION
**Pull request by Yu-Jack**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Problem:**
We strictly validate that the LUKS parameters must match the default values, which prevents users from using other parameters for encryption. This restriction doesn't make sense.

**Solution:**
Remove the value checking, and just keep the key checking.

**Related Issue:**
https://github.com/harvester/harvester/issues/6789

**Test plan:**
Follow https://docs.harvesterhci.io/v1.4/image/image-security/ and use different value to encrypt image, such as `sha512`.

## Before

image

## After

image

image

image

## Option 1

sha512 + aes-xts-plain64 + argon2i

```
node:/dev/longhorn # cryptsetup luksDump pvc-04f940dc-4a32-42b7-a025-eaa7d93b6da4
LUKS header information
Version:       	2
Epoch:         	3
Metadata area: 	16384 [bytes]
Keyslots area: 	16744448 [bytes]
UUID:          	bae0bf56-0cfb-4121-bc93-930e40b207d5
Label:         	(no label)
Subsystem:     	(no subsystem)
Flags:       	(no flags)

Data segments:
  0: crypt
	offset: 16777216 [bytes]
	length: (whole device)
	cipher: aes-xts-plain64
	sector: 512 [bytes]

Keyslots:
  0: luks2
	Key:        512 bits
	Priority:   normal
	Cipher:     aes-xts-plain64
	Cipher key: 512 bits
	PBKDF:      argon2i
	Time cost:  11
	Memory:     1048576
	Threads:    4
	Salt:       82 19 3d 94 22 10 f7 d2 8e 97 d9 1c e1 fc c9 df
	            8f 67 34 15 14 8c 7f 34 4d b8 06 4a fd 38 eb 63
	AF stripes: 4000
	AF hash:    sha512
	Area offset:32768 [bytes]
	Area length:258048 [bytes]
	Digest ID:  0
Tokens:
Digests:
  0: pbkdf2
	Hash:       sha512
	Iterations: 243176
	Salt:       55 04 17 25 b8 73 c4 cb 3b 5d 4c 92 3c c8 69 10
	            ed a4 db a4 68 a2 ea f9 1e 2b 5b 6f ea 2a 9e d1
	Digest:     e7 7b 1f 28 77 7c 10 f6 a7 c6 67 8f 47 92 ee 19
	            0e 67 d4 4e e9 63 99 4f 8f 23 9b 8c f9 1c 71 66
	            13 e3 5f fc 62 a9 5c d4 48 05 55 e7 e0 bf 03 fd
	            8a 6e f4 02 1b 13 6c eb b0 3e 42 f5 6f 14 1c c5
```

## Option 2

sha512 + aes-xts-plain + pbkdf2

```
node:/dev/longhorn # cryptsetup luksDump pvc-9e68733f-4c96-4614-8a9a-5586372b35eb
LUKS header information
Version:       	2
Epoch:         	3
Metadata area: 	16384 [bytes]
Keyslots area: 	16744448 [bytes]
UUID:          	0842afbe-5fd3-4b1e-a882-99e82ec800ab
Label:         	(no label)
Subsystem:     	(no subsystem)
Flags:       	(no flags)

Data segments:
  0: crypt
	offset: 16777216 [bytes]
	length: (whole device)
	cipher: aes-xts-plain
	sector: 512 [bytes]

Keyslots:
  0: luks2
	Key:        512 bits
	Priority:   normal
	Cipher:     aes-xts-plain
	Cipher key: 512 bits
	PBKDF:      pbkdf2
	Hash:       sha512
	Iterations: 3819948
	Salt:       fd 1a 98 59 56 e4 72 9d 1c 84 47 89 bd b7 e1 3a
	            5c 6a d9 23 ee ed 4b 04 f8 a2 ba 28 2d 94 11 3b
	AF stripes: 4000
	AF hash:    sha512
	Area offset:32768 [bytes]
	Area length:258048 [bytes]
	Digest ID:  0
Tokens:
Digests:
  0: pbkdf2
	Hash:       sha512
	Iterations: 235317
	Salt:       8a f0 64 37 31 cc 68 48 29 2f 61 57 63 04 eb ef
	            c9 8b 6a 0e a1 b7 ab 4f 46 f5 79 2b e6 3c 79 0c
	Digest:     de 63 5a 3c f5 18 0a 1e 40 b6 04 ea 85 58 3c f3
	            9a 9e e0 c0 3d d4 cb 07 56 1e 7e 9b 8d 47 76 e8
	            59 10 d2 b5 58 4c 6a 41 0c df fc da bd f3 38 03
	            48 cb 6f 50 dc 62 92 59 e2 ec 75 73 3a b3 eb 0c
```

Although we open different options for encryption, there might be some error messages which node might not support specific encryption like this.

image

### CODE REVIEW

File: pkg/webhook/resources/storageclass/validator.go [hunk 1]
```diff
@@ -196,25 +203,34 @@ func (v *storageClassValidator) validateEncryption(newObj runtime.Object) error
...
 		if !ok {
 			return werror.NewInvalidError(fmt.Sprintf("secret %s/%s is not a valid encryption secret, missing field: %s", secretNamespace, secretName, field), "")
 		}
 
-		if field == lhtypes.CryptoKeyValue {
-			invalid = string(value) == ""
-		} else {
-			invalid = string(value) != defaultValue
+		switch field {
+		case lhtypes.CryptoKeyProvider:
```
Comment (bk201) [inline pkg/webhook/resources/storageclass/validator.go:222]: @Yu-Jack as you mention at the end of the test plan:
> Although we open different options for encryption, there might be some error messages which node might not support specific encryption like this.

The check here literally validates every possible value in each field but doesn't check if the combination from all fields works. The user can only know if the encryption works after doing it. Is there a document to describe all possible combinations?

Comment (Yu-Jack) [reply pkg/webhook/resources/storageclass/validator.go:222]: Unfortunately, we don't know the possible combinations, but I created a [doc PR](https://github.com/harvester/docs/pull/682) to tell users that it eventually depends on the nodes they're using.

**Review by bk201 — APPROVED** [review]
lgtm!

**Review by WebberHuang1118 — APPROVED** [review]
LGTM, thanks.

**Pull request merged**